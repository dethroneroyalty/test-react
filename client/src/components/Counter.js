import React from "react";

export default class Counter extends React.Component {
	state = {
		count: 0
	};

	increment = () => {
		this.setState(prev => ({
			count: prev.count + 1
		}));
	};

	decrement = () => {
		this.setState(prev => ({
			count: prev.count - 1
		}));
	};

	render() {
		return (
			<div className="Counter">
				<div className="count">{this.state.count}</div>
				<Buttons increment={this.increment} decrement={this.decrement} />
			</div>
		);
	}
}

function Buttons({ increment, decrement }) {
	return (
		<div>
			<button className="increment" onClick={increment}>
				+
			</button>
			<button className="decrement" onClick={decrement}>
				-
			</button>
		</div>
	);
}
