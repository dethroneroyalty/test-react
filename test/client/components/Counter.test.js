import React from "react";
import renderer from "react-test-renderer";
import { expect } from "chai";
import { shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";
//import sinon from "sinon";
import jsdomGlobal from "jsdom-global";

import Counter from "../../../client/src/components/Counter";

describe("Counter", () => {
	let component;
	let cleanupDom;

	beforeEach(() => {
		cleanupDom = jsdomGlobal(`<div id="root"></div>`);
		component = mount(<Counter />);
	});

	afterEach(() => {
		cleanupDom();
	});

	it("initial value is 0", function() {
		expect(component.find(".count").text()).to.be.equal("0");
	});

	it("increment counter", function() {
		//const component = renderer.create(<Counter />);
		//let tree = component.toJSON();

		component.find(".increment").simulate("click");
		expect(toJSON(component)).to.matchSnapshot();
		component.find(".increment").simulate("click");
		expect(toJSON(component)).to.matchSnapshot();
		component.find(".increment").simulate("click");
		expect(toJSON(component)).to.matchSnapshot();

		expect(component.find(".count").text()).to.be.equal("3");
	});

	it("increment counter", function() {
		//const component = renderer.create(<Counter />);
		//let tree = component.toJSON();

		component.find(".decrement").simulate("click");
		expect(toJSON(component)).to.matchSnapshot();
		component.find(".decrement").simulate("click");
		expect(toJSON(component)).to.matchSnapshot();
		component.find(".decrement").simulate("click");
		expect(toJSON(component)).to.matchSnapshot();

		expect(component.find(".count").text()).to.be.equal("-3");
	});

	it("increment->decrement/decrement->increment is idempotent", () => {
		// increment->decrement
		component.find(".increment").simulate("click");
		component.find(".decrement").simulate("click");

		expect(component.find(".count").text()).to.be.equal("0");
		expect(toJSON(component)).to.matchSnapshot();

		// decrement->increment
		component.find(".decrement").simulate("click");
		component.find(".increment").simulate("click");

		expect(component.find(".count").text()).to.be.equal("0");
		expect(toJSON(component)).to.matchSnapshot();
	});
});
