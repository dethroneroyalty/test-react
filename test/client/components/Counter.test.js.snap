// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter increment counter 1`] = `
<Counter>
  <div
    className="Counter"
  >
    <div
      className="count"
    >
      1
    </div>
    <Buttons
      decrement={[Function]}
      increment={[Function]}
    >
      <div>
        <button
          className="increment"
          onClick={[Function]}
        >
          +
        </button>
        <button
          className="decrement"
          onClick={[Function]}
        >
          -
        </button>
      </div>
    </Buttons>
  </div>
</Counter>
`;

exports[`Counter increment counter 2`] = `
<Counter>
  <div
    className="Counter"
  >
    <div
      className="count"
    >
      2
    </div>
    <Buttons
      decrement={[Function]}
      increment={[Function]}
    >
      <div>
        <button
          className="increment"
          onClick={[Function]}
        >
          +
        </button>
        <button
          className="decrement"
          onClick={[Function]}
        >
          -
        </button>
      </div>
    </Buttons>
  </div>
</Counter>
`;

exports[`Counter increment counter 3`] = `
<Counter>
  <div
    className="Counter"
  >
    <div
      className="count"
    >
      3
    </div>
    <Buttons
      decrement={[Function]}
      increment={[Function]}
    >
      <div>
        <button
          className="increment"
          onClick={[Function]}
        >
          +
        </button>
        <button
          className="decrement"
          onClick={[Function]}
        >
          -
        </button>
      </div>
    </Buttons>
  </div>
</Counter>
`;

exports[`Counter increment counter 4`] = `
<Counter>
  <div
    className="Counter"
  >
    <div
      className="count"
    >
      -1
    </div>
    <Buttons
      decrement={[Function]}
      increment={[Function]}
    >
      <div>
        <button
          className="increment"
          onClick={[Function]}
        >
          +
        </button>
        <button
          className="decrement"
          onClick={[Function]}
        >
          -
        </button>
      </div>
    </Buttons>
  </div>
</Counter>
`;

exports[`Counter increment counter 5`] = `
<Counter>
  <div
    className="Counter"
  >
    <div
      className="count"
    >
      -2
    </div>
    <Buttons
      decrement={[Function]}
      increment={[Function]}
    >
      <div>
        <button
          className="increment"
          onClick={[Function]}
        >
          +
        </button>
        <button
          className="decrement"
          onClick={[Function]}
        >
          -
        </button>
      </div>
    </Buttons>
  </div>
</Counter>
`;

exports[`Counter increment counter 6`] = `
<Counter>
  <div
    className="Counter"
  >
    <div
      className="count"
    >
      -3
    </div>
    <Buttons
      decrement={[Function]}
      increment={[Function]}
    >
      <div>
        <button
          className="increment"
          onClick={[Function]}
        >
          +
        </button>
        <button
          className="decrement"
          onClick={[Function]}
        >
          -
        </button>
      </div>
    </Buttons>
  </div>
</Counter>
`;

exports[`Counter increment->decrement/decrement->increment is idempotent 1`] = `
<Counter>
  <div
    className="Counter"
  >
    <div
      className="count"
    >
      0
    </div>
    <Buttons
      decrement={[Function]}
      increment={[Function]}
    >
      <div>
        <button
          className="increment"
          onClick={[Function]}
        >
          +
        </button>
        <button
          className="decrement"
          onClick={[Function]}
        >
          -
        </button>
      </div>
    </Buttons>
  </div>
</Counter>
`;

exports[`Counter increment->decrement/decrement->increment is idempotent 2`] = `
<Counter>
  <div
    className="Counter"
  >
    <div
      className="count"
    >
      0
    </div>
    <Buttons
      decrement={[Function]}
      increment={[Function]}
    >
      <div>
        <button
          className="increment"
          onClick={[Function]}
        >
          +
        </button>
        <button
          className="decrement"
          onClick={[Function]}
        >
          -
        </button>
      </div>
    </Buttons>
  </div>
</Counter>
`;
